app/src/main/java/com/samco/trackandgraph/graphstatinput/configviews/ui/TimeHistogramConfigView.kt

```diff
@@ -54,71 +49,64 @@ fun TimeHistogramConfigView(
         viewModel.getConfigFlow().collect { onConfigEvent(it) }
     }
 
-    GraphStatDurationSpinner(
-        modifier = Modifier,
-        selectedDuration = viewModel.selectedDuration,
-        onDurationSelected = { viewModel.updateDuration(it) }
-    )
-
-    GraphStatEndingAtSpinner(
-        modifier = Modifier,
-        sampleEndingAt = viewModel.sampleEndingAt
-    ) { viewModel.updateSampleEndingAt(it) }
-
-    SpacingSmall()
-
-    Divider()
+    FormSection {
+        Column {
+            GraphStatDurationSpinner(
+                modifier = Modifier,
+                selectedDuration = viewModel.selectedDuration,
+                onDurationSelected = { viewModel.updateDuration(it) }
+            )
+
+            FormFieldSeparator()
+
+            GraphStatEndingAtSpinner(
+                modifier = Modifier,
+                sampleEndingAt = viewModel.sampleEndingAt
+            ) { viewModel.updateSampleEndingAt(it) }
+        }
+    }
 
-    SpacingLarge()
+    FormFieldSeparator()
 
-    Text(
-        modifier = Modifier.padding(horizontal = dimensionResource(id = R.dimen.card_padding)),
-        text = stringResource(id = R.string.select_a_feature),
-        style = MaterialTheme.typography.subtitle2
-    )
+    FormLabel(text = stringResource(id = R.string.select_a_feature))
 
     val featureId = viewModel.featureId
     val featureMap = viewModel.featureMap
 
     if (featureId != null && featureMap != null) {
-        TextMapSpinner(
+        FormSpinner(
             strings = featureMap,
             selectedItem = featureId,
             onItemSelected = { viewModel.updateFeatureId(it) }
         )
     }
 
-    SpacingSmall()
-
-    LabeledRow(
-        label = stringResource(id = R.string.time_window_size),
-        paddingValues = PaddingValues(start = dimensionResource(id = R.dimen.card_padding))
-    ) {
-        val stringArray = stringArrayResource(id = R.array.time_histogram_windows)
-        val timeWindows = mapOf(
-            TimeHistogramWindow.HOUR to stringArray[0],
-            TimeHistogramWindow.DAY to stringArray[1],
-            TimeHistogramWindow.WEEK to stringArray[2],
-            TimeHistogramWindow.MONTH to stringArray[3],
-            TimeHistogramWindow.THREE_MONTHS to stringArray[4],
-            TimeHistogramWindow.SIX_MONTHS to stringArray[5],
-            TimeHistogramWindow.YEAR to stringArray[6]
-        )
+    FormFieldSeparator()
 
-        TextMapSpinner(
-            strings = timeWindows,
-            selectedItem = viewModel.selectedWindow,
-            onItemSelected = { viewModel.updateWindow(it) }
-        )
-    }
+    FormLabel(text = stringResource(id = R.string.time_window_size))
 
-    SpacingSmall()
+    val stringArray = stringArrayResource(id = R.array.time_histogram_windows)
+    val timeWindows = mapOf(
```

Comment:
I know this wasn't your code but we should probably use remember { mapOf(...) } here otherwise we're potentially allocating a map every frame.